tools/config/common_settings.bzl

```diff
@@ -0,0 +1,98 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Common build setting rules for use in Bazel tools
+
+Warning: The content of this file is copied from
+https://github.com/bazelbuild/bazel-skylib/rules/common_settings.bzl
+Please do not innovate here. Keep the concepts in sync. We have a private
+copy only to prevent a looping dependency on skylib.
```

Comment:
Could you clarify what you mean by looping dependency? I thought we were keeping a copy here so that bazel doesn't have a forced dependency on skylib at all